### PR TITLE
Fixes IQ model

### PR DESCRIPTION
**Pull request by chedim**
Also contains a fix for grammar that makes `type` a valid identifier

### CODE REVIEW

File: src/main/java/com/couchbase/intellij/tree/iq/chat/ChatLinkState.java [hunk 1]
```diff
@@ -113,7 +114,7 @@ public List<ChatMessage> getChatMessages(ModelType model, ChatMessage userMessag
             // hard-coded as I couldn't find API to set it
             int maxTokens = 4097;
             var tokenizer = model.getTokenizer();
-            var chatFormatDescriptor = model.getChatFormatDescriptor();
+            var chatFormatDescriptor =  new ChatFormatDescriptor.Of(Encoding.forModel(model.modelName()), 3, 3, -1 , 3);
```
Comment (chedim) [inline src/main/java/com/couchbase/intellij/tree/iq/chat/ChatLinkState.java:117]: This was the first issue when we ran into a problem with the library.

File: src/main/java/com/couchbase/intellij/tree/iq/spi/iq/CouchbaseIQServiceProvider.java [hunk 2]
```diff
@@ -21,6 +21,10 @@
 
 public class CouchbaseIQServiceProvider implements OpenAiServiceProvider {
 
+//    public static final String MODEL = "gpt-4-1106-preview";
+//    public static final String MODEL = "text-embedding-ada-002";
+    public static final String MODEL = "gpt-4-turbo-preview";
```
Comment (chedim) [inline src/main/java/com/couchbase/intellij/tree/iq/spi/iq/CouchbaseIQServiceProvider.java:26]: the new model

File: src/main/java/com/couchbase/intellij/tree/iq/spi/iq/CouchbaseIQServiceProvider.java [hunk 3]
```diff
@@ -46,6 +50,7 @@ public OpenAiService createService(String group, OpenAISettingsState settings) {
         ObjectMapper mapper = OpenAiService.defaultObjectMapper();
         OkHttpClient client = OpenAiService.defaultClient(token, timeout)
                 .newBuilder()
+                .addInterceptor(new IQChatSettingsTransformer())
```
Comment (chedim) [inline src/main/java/com/couchbase/intellij/tree/iq/spi/iq/CouchbaseIQServiceProvider.java:53]: this is the interceptor that fixes the addition of custom fields by iq api.

**Review by deniswsrosa — APPROVED** [review]
LGTM

**Pull request merged**